queries.md

```diff
@@ -4,76 +4,96 @@
 
-<!-- Your Code Goes Here -->
+Filter: {category_code: {$type: ["null"]}}
 
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
-<!-- Your Code Goes Here -->
+Filter: {$and: [{number_of_employees: {$gt: 100}}, {number_of_employees:{$lt: 1000}}]}
+Project: {name: 1, number_of_employees: 1, _id: 0}
 
 ### 9. Order all the companies by their IPO price in a descending order.
 
-<!-- Your Code Goes Here -->
+Sort: {"ipo.valuation_amount": -1}
 
 ### 10. Retrieve the 10 companies with most employees, order by the `number of employees`
 
-<!-- Your Code Goes Here -->
+Sort: {number_of_employees: -1}
+Limit: 10
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+Filter: {founded_month: {$gt: 5}}
+Limit: 1000
 
 ### 12. All the companies founded before 2000 that have an acquisition amount of more than 10.000.000
 
-<!-- Your Code Goes Here -->
+Filter: {$and: [{founded_year: {$lt: 2000}}, {"acquisition.price_amount": {$gt: 10000000}}]}
+
 
 ### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
 
-<!-- Your Code Goes Here -->
+Filter: {"acquisition.acquired_year": {$gt: 2010}}
+Project:{name: 1, acquisition: 1, _id: 0}
+Sort: {"acquisition.price_amount": -1}
 
 ### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
 
-<!-- Your Code Goes Here -->
+Project: {name: 1, founded_year: 1, _id: 0}
+Sort: {founded_year: -1}
 
 ### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` in a descending order. Limit the search to 10 documents.
 
-<!-- Your Code Goes Here -->
+Filter: {"acquisition.acquired_day": {$lte: 7}}
+Sort: {"acquisition.acquisition_price": -1}
+Limit: 10
 
 ### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascending order.
 
-<!-- Your Code Goes Here -->
+Filter: {$and: [{category_code: "web"}, {number_of_employees: {$gt: 4000}}]}
+Sort: {number_of_employees: 1}
 
 ### 17. All the companies whose acquisition amount is more than 10.000.000, and currency is 'EUR'.
 
-<!-- Your Code Goes Here -->
+Filter: {$and: [{"acquisition.price_amount": {$gt: 10000000}}, {"acquisition.price_currency_code": "EUR"}]}
 
 ### 18. All the companies that have been acquired on the first trimester of the year. Limit the search to 10 companies, and retrieve only their `name` and `acquisition` fields.
 
-<!-- Your Code Goes Here -->
+Filter: {"acquisition.acquired_month": {$lt: 7}}
+Sort: {name: 1, acquisition: 1}
+Limit: 10
 
 ### 19. All the companies that have been founded between 2000 and 2010, but have not been acquired before 2011.
 
-<!-- Your Code Goes Here -->
```

Comment:
te dejo el hack: {$and: [{founded_year: {$gte: 2000}}, {founded_year: {$lte: 2010}}, {'acquisition.acquired_year':{$gt:2011}}]}